### DISCUSSION TITLE
Autotune: boiler still off

### DISCUSSION THREAD

**Discussion by Machaeon** [ScratMan/HASmartThermostat#187] [Q&A]
Here is my config:
 ```
 - platform: smart_thermostat
    name: Smart Boiler
    unique_id: smart_boiler
    heater: switch.boiler
    target_sensor: sensor.temperature_children
    outdoor_sensor: sensor.outside_temperature
    min_temp: 17
    max_temp: 30
    ac_mode: False
    target_temp: 23
    target_temp_step: 0.1
    cold_tolerance: 0.1
    hot_tolerance: 0
    precision: 0.1    
    keep_alive:
      seconds: 60
    away_temp: 18
    boost_temp: 24
    autotune: "ziegler-nichols"
    noiseband: 0.2
    kp: 40
    ki: 0.01
    kd: 30000
    ke: 1.5
    pwm: 00:30:00
    min_cycle_duration: 00:15:00
    min_off_cycle_duration: 00:05:00
    debug: true
```
At the moment of restart the temperature was 22.9 or 22.8. Now it has dropped to 22.6 and the boiler is still off. State at this moment:
```
hvac_modes:
  - heat
  - "off"
min_temp: 17
max_temp: 30
target_temp_step: 0.1
preset_modes:
  - none
  - away
  - boost
current_temperature: 22.6
temperature: 23
hvac_action: idle
preset_mode: none
away_temp: 18
eco_temp: null
boost_temp: 23
comfort_temp: null
home_temp: null
sleep_temp: null
activity_temp: null
control_output: 0
kp: 40
ki: 0.01
kd: 30000
ke: 1.5
pid_mode: "off"
pid_i: 0
pid_p: 0
pid_d: 0
pid_e: 0
pid_dt: 0
autotune_status: "off"
autotune_sample_time: null
autotune_tuning_rule: ziegler-nichols
autotune_set_point: null
autotune_peak_count: 0
autotune_buffer_full: 0
autotune_buffer_length: 10
friendly_name: Smart Boiler
supported_features: 17
```

Is this normal and when should my boiler switch on?

**Comment by ScratMan** [comment]
Hi, did you increase the setpoint to trigger the start of the auto-tune ?
But you simply remove the auto-tune and use the pid_off mode (hysteresis) to use manual tuning procedure described in the discussions about auto-tune.

**Comment by Machaeon** [comment]
Hello! I didn't change setpoint. Is it necessary to start autotune?
After I removed the autotune parameter and manually set the coefficients, the thermostat behaves like a generic, non PID thermostat. I see pid_mode: off in state.

**Reply by ScratMan** [reply]
It's in hysteresis mode, you can leave it like that till you can see several periods of the temperature oscillation around the setpoint. Follow the procedure below to compute the gains, set them with the set_pid_gains service and use the PID mode service to enable the PID.
https://github.com/ScratMan/HASmartThermostat/discussions/21#discussioncomment-1709437
